torch/distributed/_shard/sharded_tensor/api.py

```diff
@@ -313,6 +313,30 @@ def gather(
 
                     out_narrow_view.copy_(tensor)
 
+    def to(self,
+           device=None,
+           process_group=None):
+        if device is not None:
+            device_is_cuda = "cuda" in str(device)
+            pg_is_nccl = isinstance(self._process_group, distributed_c10d.ProcessGroupNCCL)
+
+            if device_is_cuda != pg_is_nccl:
+                assert process_group is not None, "need to pass in process_group as argument!"
```

Comment:
Hmm so I don't think we should have device specific or pg specific logic like this. We should clearly document the function mentioning that it is the users responsibility to ensure that the PG passed it works for the device passed in.